tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/symbols/FunctionTypeSymbolTest.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.semantic.api.test.symbols;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.AnnotationSymbol;
+import io.ballerina.compiler.api.symbols.FunctionSymbol;
+import io.ballerina.compiler.api.symbols.FunctionTypeSymbol;
+import io.ballerina.compiler.api.symbols.ParameterKind;
+import io.ballerina.compiler.api.symbols.ParameterSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.Project;
+import io.ballerina.tools.text.LinePosition;
+import org.ballerinalang.test.BCompileUtil;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDefaultModulesSemanticModel;
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDocumentForSingleSource;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Test cases for function type symbols.
+ *
+ * @since 2.0.0
+ */
+public class FunctionTypeSymbolTest {
+
+    private SemanticModel model;
+    private Document srcFile;
+
+    @BeforeClass
+    public void setup() {
+        Project project = BCompileUtil.loadProject("test-src/symbols/function_type_symbol_test.bal");
+        model = getDefaultModulesSemanticModel(project);
+        srcFile = getDocumentForSingleSource(project);
+    }
+
+    @Test
+    public void testFunctionType() {
+        Optional<Symbol> symbol = model.symbol(srcFile, LinePosition.from(17, 100));
+        assertTrue(symbol.isPresent());
+//        assertEquals(symbol.get().kind(), SymbolKind.VARIABLE);
+
+        TypeSymbol typeSymbol = ((FunctionSymbol) symbol.get()).typeDescriptor();
+        assertEquals(typeSymbol.typeKind(), TypeDescKind.FUNCTION);
+
+        FunctionTypeSymbol fnType = (FunctionTypeSymbol) typeSymbol;
+        Optional<List<ParameterSymbol>> params = fnType.params();
+
+        assertTrue(params.isPresent());
+        List<ParameterSymbol> paramSymbols = params.get();
+        ParameterSymbol rp = paramSymbols.get(0);
+        assertTrue(rp.getName().isEmpty());
+        assertEquals(rp.paramKind(), ParameterKind.REQUIRED);
+        assertEquals(rp.typeDescriptor().typeKind(), TypeDescKind.INT);
+
+        List<AnnotationSymbol> rpAnnots = rp.annotations();
+//        assertEquals(rpAnnots.size(), 1);
+//        assertEquals(rpAnnots.get(0).getName().get(), "v1");
```

Comment:
There are several places commented. Are they in WIP for this PR or will be fixing them separetly?

Reply:
Let's fix them separately. I created issues for them and updated the PR description with the issues.